src/Uno.Extensions.Reactive.Generator/Bindables/BindableViewModelMappingGenerator.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Uno.Extensions.Reactive.Generator;
+
+internal class BindableViewModelMappingGenerator
+{
+#pragma warning disable RS1024 // Compare symbols correctly => False positive
+	private readonly Dictionary<INamedTypeSymbol, string> _bindableVMs = new(SymbolEqualityComparer.Default);
+#pragma warning restore RS1024 // Compare symbols correctly
+	private readonly BindableGenerationContext _ctx;
+
+	public BindableViewModelMappingGenerator(BindableGenerationContext ctx)
+	{
+		_ctx = ctx;
+	}
+
+	public void Register(INamedTypeSymbol viewModelType, string bindableViewModelType)
+		=> _bindableVMs[viewModelType] = bindableViewModelType;
+
+	public (string file, string code) Generate()
+	{
+		var code = $@"#nullable enable
+			#pragma warning disable
+
+			namespace {_ctx.Context.Compilation.Assembly.Name}
```

Comment:
Is there no way to pick up the default namespace for the assembly? I'd prefer not to rely on the name of the assembly to imply namespace

Reply:
I would prefer also, but the default namespace is actually a VS property, so there is no reliable way to get (especially when building using CLI instead of VS), so I prefer to used the Assembly name. Anyway I think that the vast majority of project does not use it.

Reply:
The issue with this is that it requires all head projects to output the same assembly name (in the case that your viewmodels are in the shared/ui project). Not a major issue but imho not a good reason to force all head projects to use the same assembly name